core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java

```diff
@@ -137,7 +137,7 @@ protected EncryptedOutputFile newOutputFile(
     Preconditions.checkArgument(
         spec.isUnpartitioned() || partition != null,
         "Partition must not be null when creating output file for partitioned spec");
-    return partition == null
+    return spec.isUnpartitioned() || partition == null
```

Comment:
nit: not caused by this PR. but since we are touching the code, should we use the same style of `if-else`. 

I am fine with either if-else or the ternary operator. I know @aokolnychyi prefers if-else for multi-line statements based on a previous review.

Reply:
Great catch @stevenzwu. Consistency is king 👍🏻